typespec-extension/src/code-model-builder.ts

```diff
@@ -1530,7 +1531,8 @@ export class CodeModelBuilder {
           // use it for extensible enum
           schema = this.processChoiceSchema(knownValues, this.getName(knownValues), false);
         } else {
-          schema = this.processSchema(type.type, nameHint + "Model");
+          const schemaNameHint = pascalCase(getModelNameForProperty(type)) + pascalCase(nameHint);
+          schema = this.processSchema(type.type, schemaNameHint);
```

Comment:
This part is pretty tricky. In future, we'd switch to name from TCGC.

For now, make it works in java emitter, and output name similar to .NET